src/Components/profile/Profile.js

```diff
@@ -0,0 +1,227 @@
+  const [eventsList, setEventList] = useState(
+    JSON.parse(localStorage.getItem("Events"))
+  );
+
+  const [pageNumber, setPageNumber] = useState(0);
+  const eventsPerPage = 6;
+  const pageCount = Math.ceil(eventsList.length / eventsPerPage);
+  const pagesVisited = pageNumber * eventsPerPage;
+  const changePage = ({ selected }) => {
+    setPageNumber(selected);
+  };
+
+  useEffect(() => {
+    paintAvatarAndName(setTextAvatar, setUserName);
+    drawUserInformation();
+  }, []);
+
+  useEffect(() => {
+    const informationUser = getFromLocalStorage();
+    if (informationUser && !informationUser.isLoggedIn) {
+      setSignOut(true);
+    }
+  }, []);
+
+  const drawUserInformation = () => {
+    const userInformation = getFromLocalStorage();
+
+    let name = userInformation.name;
+    let lastName = userInformation.lastName;
+    let email = userInformation.email;
+
+    setNameUser(name);
+    setLastNameUser(lastName);
+    setEmailUser(email);
+  };
+
+  if (eventsList === null) {
+    return;
+  }
+
+  const displayEvents = eventsList
+    .slice(pagesVisited, pagesVisited + eventsPerPage)
+    .map((element, index) => {
+      if (element.stateEvent === "EDIT" || element.stateEvent === "LEAVE") {
+        return (
+          <div
+            key={index}
+            className={
+              viewEvents
+                ? `element-${index} element`
+                : `element-${index} element-column`
+            }
+            onClick={(e) =>
+              showDetailEventClicked(e, setGoToDetailEvent, setEventClicked)
+            }
+          >
+            <div className="date-time-container">
+              <spam className="date">{element.date}</spam>
+              <span className="dash">-</span>
+              <spam className="time">{element.time}</spam>
+            </div>
+            <h4 className="title-event">{element.nameEvent}</h4>
+            <p className="p-host">{element.host}</p>
+            <p className="description-event">{element.descriptionEvent}</p>
+            <div className="button-attendees-capacity-container ">
+              <span className="attendees">
+                <PersonIcon className={viewEvents ? "" : "icon-hide-person"} />
+                <span>{element.attendees}</span>
+                <span className="of-text">of</span>
+                <span>{element.capacity}</span>
+              </span>
+              <Button
+                variant="contained"
+                className={`button-event ${element.id}`}
+                onClick={(e) => {
+                  handleButtonEvent(
+                    e,
+                    setGoToEditEvent,
+                    setEventToEdit,
+                    eventToEdit,
+                    eventsList,
+                    setEventList
+                  );
+                }}
+              >
+                {element.stateEvent}
+              </Button>
+            </div>
+          </div>
+        );
+      }
```

Comment:
we need to create a component for this, to avoid adding all these code in here

Reply:
I created a drawEvents.js file since I also have to do the same for the dashboard, so what I did was save it in a function and import it